packages/extension/src/view/devtools/components/cookies/cookiesListing/cookieTableContainer/editableComponents/editableTextInput.tsx

```diff
@@ -0,0 +1,164 @@
+
+interface EditableTextInputProps {
+  info: InfoType;
+  changedKey: string;
+  cookieKey?: string | null;
+  rowHighlighter?: (value: boolean, cookieKey: string) => void;
+}
+
+const EditableTextInput = ({
+  info,
+  changedKey,
+  cookieKey,
+  rowHighlighter = noop,
+}: EditableTextInputProps) => {
+  const [editing, setEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [localValue, setLocalValue] = useState<string>(info as string);
+  const { modifyCookie } = useCookieStore(({ actions }) => ({
+    modifyCookie: actions.modifyCookie,
+  }));
+
+  const handleDoubleClick = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (event.detail === 2) {
+        setEditing(true);
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (editing) {
+      inputRef.current?.select();
+      inputRef.current?.setSelectionRange(0, -1);
+    }
+  }, [editing]);
+
+  const handleClickOutside = useCallback(
+    async (event: MouseEvent) => {
+      if (
+        inputRef.current &&
+        !inputRef.current.contains(event.target as Node)
+      ) {
+        setEditing(false);
+        if (localValue !== info && cookieKey) {
+          const result = await modifyCookie(
+            cookieKey,
+            changedKey,
+            localValue.trim(),
+            changedKey === 'name' ? (info as string) : null
+          );
+
+          if (!result) {
+            rowHighlighter(true, cookieKey);
+            setLocalValue(info as string);
+          } else {
+            rowHighlighter(false, cookieKey);
+          }
+        }
+      }
+    },
+    [localValue, info, cookieKey, modifyCookie, changedKey, rowHighlighter]
+  );
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      setLocalValue(event.target.value.trim());
+    },
+    []
+  );
+
+  const handleKeyDown = useCallback(
+    async (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event?.key === 'Enter') {
+        setEditing(false);
+        if (localValue !== info && cookieKey) {
+          const result = await modifyCookie(
+            cookieKey,
+            changedKey,
+            localValue,
+            changedKey === 'name' ? (info as string) : null
+          );
+
+          if (!result) {
+            rowHighlighter(true, cookieKey);
+            setLocalValue(info as string);
+          } else {
+            rowHighlighter(false, cookieKey);
+          }
+        }
+      }
+      if (event?.key === 'Escape') {
+        setEditing(false);
+      }
+    },
+    [changedKey, cookieKey, info, localValue, modifyCookie, rowHighlighter]
+  );
+
+  useEffect(() => {
+    setLocalValue(info as string);
+    setEditing(false);
+  }, [info, cookieKey]);
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
```

Comment:
I guess we can remove this comment, the code explains itself.

Reply:
Added in [8156084](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/278/commits/8156084639e3a4a02224f51f9a795f75e05175ff)